envoyproxy/envoy
# Fix Windows compilation of test sources

Description:

This patch is extracted from https://github.com/envoyproxy/envoy/pull/10542 and must be preceded by that merge in order for all enabled tests to compile successfully on Windows.

This patch should still be correct in the interim, and should effectively be a no-op on Linux.

- Remove erroneous test from PR #9964
  - Test macro missing the second argument (for test name),
    causing compilation on Windows to fail. It appears to
    have no purpose, copy-paste error?
- more involved compilation failures in test sources are
  excluded with skip_on_windows tag
  - involving tests to do with features that don't exist for
    windows (signals, etc.)
  - sh_test rules do not currently work on Windows, waiting on a Bazel
    fix
- Individually exclude tests in non-compiling extensions
  with skip_on_windows
- subset of failing tests tagged with fails_on_windows
  - tests that time out
  - tests that are known to fail
  - non-exhaustive list, additional failing tests will be
    fixed/tagged in subsequent efforts
- various fixes for Windows compilation
  - fix os_fd_t type in mocks
  - ensure consistent integer types
  - proliferate use of OsSysCalls in tests
  - exclude signals tests
  - add _set_invalid_parameter_handler to test main to ensure runtime
    does not abort when functions are called with invalid parameters
  - print warning about tcpdump not currently being enabled on Windows
  - matchers_test.cc: do not return address of literal
  - test/extensions/filters/network/direct_response/BUILD: skip test with
    constructor+lambda that does not compile on Windows
  - codec_impl_test.cc: MSVC pre-processing does not like the multline
    string passed to EXPECT_EQ macro, pull into intermediate variable

Co-authored-by: Sunjay Bhatia <sbhatia@pivotal.io>
Co-authored-by: William A Rowe Jr <wrowe@pivotal.io>
Signed-off-by: Sunjay Bhatia <sbhatia@pivotal.io>
Signed-off-by: William A Rowe Jr <wrowe@pivotal.io>

Risk Level: Low
Testing: Local on msvc-cl, linux-gcc
Docs Changes: n/a
Release Notes: n/a
[Optional Fixes #Issue]
[Optional Deprecated:]
